_posts/2018-06-11-using-strimzi-with-local-storage.md

```diff
@@ -0,0 +1,134 @@
+
+The ephemeral storage behaves differently.
+An ephemeral volume is usually just a directory somewhere in the host operating system of your node.
+It behaves very similarly to a temporary directory.
+When your Kafka broker is scheduled to a node, it can use the ephemeral volume and store data in it.
+But when the node crashes or the Pod is rescheduled, it will receive a new ephemeral disk which will not contain the previously stored data.
+It will start with an empty volume.
+It will therefore need to resync all its messages from one of the other replicas.
+Depending on the size of your partitions, this could take a very long time.
+Even if the original node was just restarted, the data stored in ephemeral volume usually do not survive the restart.
+The Pod with your broker will also usually not wait for the node to come back online and it will be started on another node.
+
+![Ephemeral storage during node crash]({{ "/assets/2018-06-11-using-strimzi-with-local-storage-ephemeral-storage.png" | absolute_url }})
+
+The choice between persistent network attached storage and ephemeral local storage is hard.
+But often - especially with large brokers storing terrabytes of data - the persistent network attached storage can be seen as the better option.
+But in the last versions, Kubernetes added support for the _Local_ storage type.
+How good match is _Local_ storage for Kafka?
+
+# Local persistent storage
+
+Local storage offers persistent volumes which are tied to a particular host.
+It is currently in a _beta_ version.
+Features in _beta_ release usually tend to be fairly stable and work very well.
+But there is no guarantee that they will not change before graduating from _beta_.
+If you want to find out more about the Local storage type and how to use it, [this post](https://kubernetes.io/blog/2018/04/13/local-persistent-volumes-beta/) might be a good way to get started.
+
+Local storage, to some extent, combines the best of both worlds.
+It is using storage which is locally attached to the node but it is persistent.
+So when the node restarts, the data which were stored on it survive.
+Additionally, since the Local volume is tied to a particular node, it will make sure that the Pod with your Kafka broker will not be rescheduled to another host.
+Kubernetes will wait until your node is back online and only then schedule the Kafka broker on it.
+When the broker is started, it will find the volume in the same state as it was left.
+It has to resync only the records which it missed during the failover - this is exactly the same as when you use persistent network attached storage.
+
+![Local persistent storage during node restart]({{ "/assets/2018-06-11-using-strimzi-with-local-storage-local-storage.png" | absolute_url }})
+
+Now ... what if your node is gone and will never return?
+In that case, you will need to make sure that the broker is scheduled on another node.
```

Comment:
How the user can achieve that? Should we explain here?

Reply:
Basically, you delete the PVC and the Pod and wait until a new one is scheduled. Not sure whether we will come up with some more elegant solution. That is why I didn't went into the details. Maybe in next blog post :-)